test/test_test.dart

```diff
@@ -139,6 +145,6 @@ void main() {
       sawError = true;
     }
 
-    expect(sawError, isTrue);
+    expect(sawError, isFalse);
```

Comment:
While your test looks good to check if the fix works, you're also effectively removing an existing test that checks that failures can be properly detected.  Can you make this a separate test so that we don't lose coverage on the old functionality?

Reply:
The original intent of the test was to check that when a failure (`severe` message) occurs in the `check` phase, that the test correctly fails.  The new test you'd like to add for this change is what you have currently implemented in this PR, where if it occurs once it passes, if it occurs twice it fails.

You'll want to create a copy of both the test `test('Failure during check causes test failure'`, maybe named something like `test('Check is only called once on components directly under Test'`, as well as a copy of the `class FailingSubComponent extends Component` and `class CheckPhaseFailureTest extends Test` that do specifically what you're trying to do.

The point is that the original intent of the test is lost/replaced by your change, whereas we actually want to add a new test which tests a different thing.